fast64_internal/f3d/f3d_material_helpers.py

```diff
@@ -0,0 +1,36 @@
+import bpy
+
+
+class F3DMaterial_UpdateLock:
+    material: bpy.types.Material = None
+
+    def __init__(self, material: bpy.types.Material):
+        self.material = material
+        if self.mat_is_locked():
+            # Disallow access to locked materials
+            self.material = None
+
+    def __enter__(self):
+        if self.mat_is_locked():
+            return None
+
+        self.lock_material()
+        return self.material
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        self.unlock_material()
+        if exc_value:
+            print("\nExecution type:", exc_type)
+            print("\nExecution value:", exc_value)
+            print("\nTraceback:", traceback)

```

Comment:
Did you test what this prints?
I would expect using `traceback` or `sys.excepthook` to behave better

Reply:
Lemme check on this, I had found an `__exit__` example and this is what it did

Reply:
![image](https://user-images.githubusercontent.com/79979276/181500066-1ce5fe64-636b-4f88-8e25-40b4c7dcf4b4.png)

i think this outputs nicely!